release-notes/2023/v1.9.0.md

```diff
@@ -0,0 +1,397 @@
+
+After:
+
+``` java
+PickUpOutcome outcome = workRegistry.pickUp(index, currentNode);
+if (outcome.hasSession()) { // Check if shard is picked
+    // ...
+    workRegistry.release(outcome.getSession()); // Release shard.
+}
+```
+
+Also, the new API allows getting the `WorkerId` of the worker who owns the session in case if the shard is already picked by someone else and the `Timestamp` when the shard was picked:
+``` java
+PickUpOutcome outcome = workRegistry.pickUp(index, currentNode);
+if (outcome.hasAlreadyPickedBy()) {
+    WorkerId worker = outcome.getAlreadyPicked().getWorker();
+    Timestamp whenPicked = outcome.getAlreadyPicked().getWhenPicked();
+    // ...
+}
+```
+
+### Other changes
+
+
+- **Custom `Executor` for `SystemSettings` ([#1448](https://github.com/SpineEventEngine/core-java/pull/1448))**
+
+  Now, `SystemSettings` allows customizing an `Executor` to post the system events in parallel. This provides an opportunity to improve the control over the available CPU resources on a server instance.
+  
+    ``` java
+    var builder = BoundedContextBuilder.assumingTests();
+    var executor = ...;
+    
+    builder.systemSettings()
+        .enableParallelPosting()
+        .useCustomExecutor(executor);
+    ```
+
+- **Customization of gRPC `Server` via `GrpcContainer` ([#1454](https://github.com/SpineEventEngine/core-java/pull/1454))**
+
+  It is now possible to access an underlying instance of `Server`'s builder when configuring the `GrpcContainer`:
+  
+  ``` java
+  GrpcContainer.atPort(1654)
+               // `server` is an instance of `io.grpc.ServerBuilder`.
+               .withServer((server) -> server.maxInboundMessageSize(16_000_000))
+               // ...
+               .build();
+  ```
+
+  This API is **experimental** and may change in future versions of Spine.
+
+
+- **Thorough copying of Bounded Contexts by `BlackBoxContext`'s builder ([#1495](https://github.com/SpineEventEngine/core-java/pull/1495))**
+
+  Previously, the` BlackBoxContext` instances were built on top of `BoundedContextBuilder`s by copying the internals of the latter builder. However, not all the parts were copied properly.
+  
+  This release improves the copying by including more pieces from the source `BoundedContextBuilder`. In particular, all changes made to `BoundedContextBuilder.systemSettings()` are now transferred as well.
+
+
+- **Custom handlers for failed delivery of a signal ([#1496](https://github.com/SpineEventEngine/core-java/pull/1496))**
+
+  Now, the Delivery API allows to subscribe for any failures which occur during the reception of each signal. Additionally, end-users may now choose the way to handle the reception failures in terms of action in respect to the InboxMessage of interest.
+  
+  Out-of-the-box, end-users are provided with two options:
+  
+  * mark the `InboxMessage` as delivered — so that it does not block further delivery of messages;
+    * repeat the dispatching of `InboxMessage` in a synchronous manner.
```

Comment:
```suggestion
  * mark the `InboxMessage` as delivered — so that it does not block further delivery of messages;
  * repeat the dispatching of `InboxMessage` in a synchronous manner.
```